src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStarImportCheckTest.java

```diff
@@ -69,6 +76,33 @@ public void testExcludes()
                 expected2);
     }
 
+    /**
+     * As we cannot reproduce situation when we set excludes several times during one run
+     */
+    @Test
+    public void testExcludesCleared() throws Exception {
+        final File input = new File(getPath("InputAvoidStarInputTestExcludes.java"));
+        final DetailAST firstImportAst = TestUtils.parseFile(input).getNextSibling();
+        final DetailAST secondImportAst = firstImportAst.getNextSibling();
+        final AvoidStarImportCheck check = new AvoidStarImportCheck();
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(AvoidStarImportCheck.class);
+        check.configure(checkConfig);
+
+        check.setExcludes("java.io.*");
+        check.visitToken(firstImportAst);
+        check.setExcludes("java.lang.*");
+        check.visitToken(firstImportAst);
+        check.visitToken(secondImportAst);
```

Comment:
I need good reason to for this type of tests.

Reply:
There was comment over the method with short description: 
> As we cannot reproduce situation when we set excludes several times during one run

Reply:
@Nimfadora ,

> This method is not used in the project

setExcludes method for Check property - checkstyle.sourceforge.net/config_imports.html#AvoidStarImport `excludes`

> As we cannot reproduce situation when we set excludes several times during one run

please try to call  `check.configure(checkConfig);` two times with different checkConfig, validation should be done on config that you did second time.

Reply:
@romani when I call `check.configure(checkConfig)` two times on different configs it just checks the same input by two different configs and the state is clear when second config check starts.

Reply:
yes, 

Reason: checkstyle do not allow to make a tree of modules manually , all should be done by configuration declaration. Two configurations result in two Check instances. NO way to reconfigure Check after execution is validation is written in non hack way (by c-tor of Check and manual configuration of it). We can execute validation by whole Checker(with all childs) not by usage of certain Check. So there is way to reproduce this mutation. 

So lets remove this line completely from Check.

Reply:
have I understood you right: we delete [line](https://nimfadora.github.io/pr/com.puppycrawl.tools.checkstyle.checks.imports/AvoidStarImportCheck.java.html#org.pitest.mutationtest.report.html.SourceFile@29cfaa8b_117) `excludes.clear()` from `AvoidStarImportCheck` and the same thing with `IllegalImportCheck`, where we delete [line](https://nimfadora.github.io/pr/com.puppycrawl.tools.checkstyle.checks.imports/IllegalImportCheck.java.html#org.pitest.mutationtest.report.html.SourceFile@3115f9c9_118) `illegalClassesRegexps.clear();`

Reply:
@Nimfadora , yes, just remove that unnecessary collections cleaups.